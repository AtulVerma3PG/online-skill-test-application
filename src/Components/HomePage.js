import React, { Component } from "react";
import { DropdownButton, Dropdown } from "react-bootstrap";
import _ from "lodash";
import "bootstrap/dist/css/bootstrap.min.css";

const skillLevels = ["Level 1", "Level 2", "Level 3"];

class HomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      IsLevelFilled: false,
      IsAggreed: false,
      IsQuizSubmitted: false,
      firstName: "",
      lastName: "",
      email: "",
      password: "",
      gender: "",
      isRegistered: false,
      skillLevel: "Select the skill level",
    };
  }

  /**
   * handling back button click and updating the state for the registration detailsS
   */
  componentDidMount() {
    const { history } = this.props;
    window.onpopstate = () => {
      history.push({
        pathname: "/LandingPage",
        state: this.state,
      });
    };
    this.setState({
      firstName: _.get(this.props, "location.state.firstName"),
      lastName: _.get(this.props, "location.state.lastName"),
      password: _.get(this.props, "location.state.password"),
      email: _.get(this.props, "location.state.email"),
      gender: _.get(this.props, "location.state.gender"),
      isRegistered: _.get(this.props, "location.state.isRegistered"),
    });
  }

  /**
   * Clear local storage and start questionaire
   */
  startTestHandler = () => {
    const { history } = this.props;
    localStorage.clear();
    history.push({
      pathname: "/Questionaire",
      state: { ...this.state },
    });
  };

  /**
   * Handle skill level selected
   *
   * @param {string} eventKey Skill level selected
   * @param {*} event
   */
  handleSelect(eventKey) {
    this.setState({ skillLevel: eventKey, IsLevelFilled: true });
  }

  /**
   * Handle skill level selected
   *
   * @param {string} eventKey Skill level selected
   * @param {*} event
   */
  handleAgreement() {
    const { IsAggreed } = this.state;
    this.setState({ IsAggreed: !IsAggreed });
  }

  render() {
    const { IsLevelFilled, skillLevel, IsAggreed } = this.state;
    return (
      <div>
        <div>
          <h1>Skill Examination</h1>
          <br />
          <h2>Examination Instructions</h2>
          <ul className="list-group">
            <li className="list-group-item">
              A start button would start the test with specific time
            </li>
            <li className="list-group-item">
              Candidates can view one question per page and can navigate
              next/previous.
            </li>
            <li className="list-group-item">
              Question may be a single or multiple choice.
            </li>
            <li className="list-group-item">
              Test would finish automatically after the timer ends.
            </li>
            <li className="list-group-item">
              The time limit for the test is 10 minutes.
            </li>
          </ul>
        </div>
        <br />
        <br />
        <br />

        <div>
          <DropdownButton
            title={skillLevel}
            id="selectLevel"
            onSelect={(eventKey) => this.handleSelect(eventKey)}
          >
            {skillLevels.map((opt) => (
              <Dropdown.Item as="button" eventKey={opt} key={opt}>
                {opt}
              </Dropdown.Item>
            ))}
          </DropdownButton>
          <div className="checkbox">
            <label
              className="checkbox-inline"
              htmlFor="{choice}"
              style={{ fontWeight: "bold" }}
            >
              <input
                type="checkbox"
                value="agreement"
                onChange={() => this.handleAgreement()}
              />
              I agree to the Terms and Conditions
            </label>
            <br />
          </div>
        </div>
        <br />
        <br />
        {IsLevelFilled && IsAggreed && (
          <div>
            <button
              type="button"
              className="btn btn-primary btn-lg"
              onClick={this.startTestHandler}
            >
              Start Test
            </button>
          </div>
        )}
      </div>
    );
  }
}

HomePage.propTypes = {
  history: Object.isRequired,
};

export default HomePage;
