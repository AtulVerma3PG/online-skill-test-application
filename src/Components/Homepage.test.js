/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import HomePage from "./HomePage";

const state = {
  firstName: "",
  lastName: "",
  email: "",
  password: "",
  gender: "",
  isRegistered: false,
  formErrors: {
    firstName: "",
    lastName: "",
    gender: "",
    email: "",
    password: "",
  },
  firstNameValid: false,
  lastNameValid: false,
  emailValid: false,
  passwordValid: false,
  genderValid: false,
  formValid: false,
}

configure({adapter: new Adapter()});
test("HomePage Snapshot test", () => {
  localStorage.setItem("userDetails", JSON.stringify(state));
  const component = renderer.create(
    <HomePage />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  localStorage.removeItem("userDetails");
});
