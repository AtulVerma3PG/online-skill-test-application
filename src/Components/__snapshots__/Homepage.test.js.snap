// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Snapshot test 1`] = `
<div>
  <nav
    className="menu-container-nav"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
  </nav>
  <div
    className="container"
  >
    <div
      className="home"
    >
      <h1>
        Skill Examination
      </h1>
      <br />
      <div
        className="quizinstructions"
      >
        <h4>
          Examination Instructions
        </h4>
        <ol
          type="1"
        >
          <li>
            A start button would start the test with specific time
          </li>
          <li>
            Candidates can view one question per page and can navigate next/previous.
          </li>
          <li>
            Question may be a single or multiple choice.
          </li>
          <li>
            Test would finish automatically after the timer ends.
          </li>
          <li>
            The time limit for the test is 10 minutes.
          </li>
        </ol>
      </div>
      <div
        className="button-center"
      >
        <div
          className="select dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-primary"
            disabled={false}
            id="selectLevel"
            onClick={[Function]}
            type="button"
          >
            Select the skill level
          </button>
        </div>
        <div
          className="checkbox"
        >
          <label
            className="checkbox-inline"
            htmlFor="{choice}"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            <input
              onChange={[Function]}
              type="checkbox"
              value="agreement"
            />
            I agree to the Terms and Conditions
          </label>
          <br />
        </div>
      </div>
    </div>
  </div>
  <p
    className="footer"
  >
    Copyright Â© 2020 , All Rights Reserved.
  </p>
</div>
`;
